diff --git a/BUILD.bazel b/BUILD.bazel
diff --git a/BUILD.bazel b/BUILD.bazel
index 396537f..11b5eda 100644
--- a/BUILD.bazel
+++ b/BUILD_new.bazel
@@ -67,9 +67,10 @@ cc_library(
     hdrs = [
         "include/pthreadpool.h",
     ],
-    copts = [
-        "-std=c11",
-    ] + select({
+    copts = select({
+        "@bazel_tools//src/conditions:windows": ["/std:c11"], 
+        "//conditions:default": ["-std=c11"],
+    }) + select({
         ":optimized_build": ["-O2"],
         "//conditions:default": [],
     }) + select({
